client/src/domain/Community/Community.js

```diff
@@ -0,0 +1,89 @@
+import "./Community.css";
+import { Link } from "react-router-dom";
+
+import CommunityFallbackProfilePic from "../../assets/happy-students.jpg";
+
+import { FaRegEdit } from "react-icons/fa";
+import { FaTrashAlt } from "react-icons/fa";
+import axios from "axios";
+
+import CommunitiesContext from "../../contexts/CommunitiesContext";
+import { useContext } from "react";
+
+import { useHistory } from "react-router";
+
+// TODO: Get rid of that conditional wrapper again - shoudl be possible to just set an onLick-handler on the section-tag itself and basrta kanzla!
+const ConditionalWrapper = ({ condition, wrapper, children }) =>
+  condition ? wrapper(children) : children;
+
+export default function Community({ community, as }) {
+  const renderAsListItem = as && as === "ListItem";
+  const communityProfilePicImgSrc = community?.picture
+    ? community?.picture
+    : CommunityFallbackProfilePic;
+
+  const { resources, setResources } = useContext(CommunitiesContext);
+
+  const history = useHistory();
+
+  const removeResource = (id) => {
+    axios
+      .delete(`api/communities/${id}`)
+      .then((res) => {
+        setResources(
+          resources.filter((resource) => {
+            return resource._id !== id;
+          })
+        );
+        // history.push("/");
+        history.goBack();
+      })
+      .catch((err) => {
+        console.log(
+          "Failed to delete community resource - something went wrong: ",
+          err
+        );
+      });
+  };
+
+  return (
+    <section
+      className={`Resource Community Community--${community.type} ${
+        renderAsListItem && "Resource__ListItem Community__ListItem"
+      }`}
+      key={community.id}
```

Comment:
Why do you set a `key` here?

Reply:
Because Community is conditionally rendered as list item - or not  (this the conditional wrapper) - but I'm not sure if I'm gonna keep that - since my community-show-view is so different from the usual crud-interface-logic that it would make little sense to let community.js play both roles - on the other hand: it wouldn't be any problem at all to conditionally render the complex current community view from there as well - so ... let's see how it plays out